chainer/computational_graph.py

```diff
@@ -0,0 +1,201 @@
+import heapq
+
+from chainer import function
+from chainer import variable
+
+
+class DotNode(object):
+    """Node of computational graph, with utilities for dot language
+
+    This class represents a node of computational graph,
+    with some utilities for dot language
+    """
+
+    def __init__(self, node):
+        """Initialize DotNode
+
+        Args:
+            node: :class: `Variable` object or :class: `Function` object
+        """
+
+        assert isinstance(node, variable.Variable) or\
+            isinstance(node, function.Function)
+        self.node = node
+        self.id_ = id(node)
+        self.attribute = {
+            "label": self.node.label,
+            "shape": self._shape()
+        }
+
+    def _shape(self):
+        """Return shape type of node
+
+        """
+
+        if isinstance(self.node, variable.Variable):
+            return "oval"
+        elif isinstance(self.node, function.Split):
+            return "hexagon"
+        else:
+            return "box"
+
+    @property
+    def label(self):
+        """Return label that represents its property
+
+        Returns:
+            string: label that represents its id and attributes
+        """
+
+        attributes = ["%s=\"%s\"" % (k, v) for (k, v)
+                      in self.attribute.items()]
+        return "%s [%s];" % (self.id_, ",".join(attributes))
+
+
+class ComputationalGraph(object):
+    """Class that represents computational graph
+
+    .. note::
+
+      We assume that computational graph is directed and is a DAG.
```

Comment:
-> We assume that computational grpah is directed and acyclic.
(DAG includes "directed").